mifosng-android/src/main/java/com/mifos/mifosxdroid/online/sharingaccount/SharingAccountFragment.java

```diff
@@ -0,0 +1,421 @@
+    }
+
+    @OnClick(R.id.tv_application_date)
+    public void setTvApplicationDate() {
+        isApplicationDate = true;
+        mfDatePicker.show(getActivity().getSupportFragmentManager(), FragmentConstants
+                .DFRAG_DATE_PICKER);
+    }
+
+    @Override
+    public void onDatePicked(String date) {
+
+        if (isSubmissionDate) {
+            tvSubmissionDate.setText(date);
+            submissionDate = DateHelper.getDateAsStringUsedForCollectionSheetPayload
+                    (date).replace("-", " ");
+            isSubmissionDate = false;
+        }
+
+        if (isApplicationDate) {
+            tvApplicationDate.setText(date);
+            applicationDate = DateHelper.getDateAsStringUsedForCollectionSheetPayload
+                    (date).replace("-", " ");
+            isApplicationDate = false;
+        }
+    }
+
+    @Override
+    public void showSharingAccountTemplate(List<ProductSharing> productSharing,
+                                           List<ChargeOptions> chargeOptions) {
+        mProductSharings = productSharing;
+        mChargeOptions = chargeOptions;
+
+        mListSharingProductsNames.addAll(mSharingAccountPresenter
+                .filterSharingProductsNames(productSharing));
+        mSharingProductsAdapter.notifyDataSetChanged();
+
+        mChargesNames.addAll(mSharingAccountPresenter
+                .filterChargeOptionNames(chargeOptions));
+        mChargesAdapter.notifyDataSetChanged();
+    }
+
+    @Override
+    public void showSavingsAccountTemplateByProduct(SharingAccountsTemplate
+                                                                sharingAccountsTemplate) {
+        mSharingProductsTemplateByProductId = sharingAccountsTemplate;
+
+
+        mListLockinPeriodOptions.clear();
+        mListLockinPeriodOptions.addAll(mSharingAccountPresenter
+                .filterLockinPeriodFrequencyTypesName(sharingAccountsTemplate.
+                        getLockinPeriodFrequencyTypeOptions()));
+        mLockinPeriodAdapter.notifyDataSetChanged();
+
+        mListMinActivePeriodOptions.clear();
+        mListMinActivePeriodOptions.addAll(mSharingAccountPresenter
+                .filterMinimumActivePeriodFrequencyTypesNames(sharingAccountsTemplate.
+                        getMinimumActivePeriodFrequencyTypeOptions()));
+        mMinActivePeriodAdapter.notifyDataSetChanged();
+
+        mListSavingsAccountOptions.clear();
+        mListSavingsAccountOptions.addAll(mSharingAccountPresenter
+                .filterSavingProductsNames(sharingAccountsTemplate.getClientSavingsAccounts()));
+        mSavingsAccountAdapter.notifyDataSetChanged();
+        showDefaultValues();
+    }
+
+    @Override
+    public void showSharingAccountCreatedSuccessfully(ShareResponse shareResponse) {
+        Toast.makeText(getActivity(),
+                getResources().getString(R.string.sharing_account_submitted_for_approval),
```

Comment:
yes It was added in API 1 : https://developer.android.com/reference/android/content/Context.html#getString(int)